testutil/gen.go

```diff
@@ -0,0 +1,81 @@
+package testutil
+
+import (
+	"regexp"
+	"strings"
+	"testing"
+
+	"github.com/genjidb/genji/document"
+	"github.com/genjidb/genji/expr"
+	"github.com/genjidb/genji/query"
+	"github.com/genjidb/genji/sql/parser"
+	"github.com/stretchr/testify/require"
+)
+
+type DocsStream struct {
+	*parser.Parser
+	env *expr.Environment
+}
+
+func (ds *DocsStream) Next() (*document.Value, error) {
+	exp, err := ds.Parser.ParseDocument()
+	if err != nil {
+		return nil, err
+	}
+
+	v, err := exp.Eval(ds.env)
+	if err != nil {
+		return nil, err
+	}
+
+	return &v, nil
+}
+
+func ParseDocsStream(stream string) *DocsStream {
+	p := parser.NewParser(strings.NewReader(stream))
+	env := expr.NewEnvironment(document.NewFieldBuffer())
+
+	return &DocsStream{p, env}
+}
+
+func RequireErrorMatch(t *testing.T, regxp string, err error) {
```

Comment:
These functions are quite trivial, but they do simplify the template, as they allow to not have to guess if the `testutil` package needs to be imported or not.

Reply:
This is not necessary if you use https://pkg.go.dev/golang.org/x/tools/imports. This is basically `goimports` as a lib

Here is an example:

```golang
	// generate code
	err = t.ExecuteTemplate(&buf, ...)
	if err != nil {
		return err
	}

	// format using goimports
	output, err := imports.Process("", buf.Bytes(), &imports.Options{
		TabWidth:   8,
		TabIndent:  true,
		Comments:   true,
	})
	if err != nil {
		return err
	}
```